core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsTests.java

```diff
@@ -941,6 +956,47 @@ public void testNodeNotReachableFromMaster() throws Exception {
         ensureStableCluster(3);
     }
 
+    @Test
+    public void testIndexImportedFromDataOnlyNodesIfMasterLostDataFolder() throws Exception {
+        // test for https://github.com/elastic/elasticsearch/issues/8823
+        configureCluster(2, 1);
+        String masterNode = internalCluster().startMasterNode(Settings.EMPTY);
+        internalCluster().startDataOnlyNode(Settings.EMPTY);
+
+        ensureStableCluster(2);
+        assertAcked(prepareCreate("index").setSettings(Settings.builder().put("index.number_of_replicas", 0)));
+        index("index", "doc", "1", jsonBuilder().startObject().field("text", "some text").endObject());
+        ensureGreen();
+
+        internalCluster().restartRandomNode(new InternalTestCluster.MasterNodePredicate(masterNode), new InternalTestCluster.RestartCallback() {
+            public boolean clearData(String nodeName) {
+                return true;
+            }
+        });
+
+        ensureGreen("index");
+        assertTrue(client().prepareGet("index", "doc", "1").get().isExists());
+    }
+
+    // tests if indices are really deleted even if a master transition inbetween
+    @Ignore("https://github.com/elastic/elasticsearch/issues/11665")
+    @Test
+    public void testIndicesDeleted() throws Exception {
+        configureCluster(3, 2);
+        String masterNode1 = internalCluster().startMasterNode(Settings.EMPTY);
+        internalCluster().startMasterNode(Settings.EMPTY);
+        String dataNode = internalCluster().startDataOnlyNode(Settings.EMPTY);
+        assertAcked(prepareCreate("test"));
+        ensureYellow();
+
+        NetworkPartition networkPartition = new NetworkUnresponsivePartition(masterNode1, dataNode, getRandom());
+        internalCluster().setDisruptionScheme(networkPartition);
+        networkPartition.startDisrupting();
+        internalCluster().client(masterNode1).admin().indices().prepareDelete("test").setTimeout("1s").get();
+        internalCluster().restartRandomNode(new InternalTestCluster.MasterNodePredicate(masterNode1), InternalTestCluster.EMPTY_CALLBACK);
```

Comment:
I cannot, at least not without changing more. The disruption sometimes prevents the master to be stopped successfully because the cluster needs a client to figure out the master name and that might not have access to the master node, because of the disruption.